stackhpc_monasca_agent_plugins/checks/slurm.py

```diff
@@ -0,0 +1,134 @@
+# Copyright (c) 2018 StackHPC Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+import copy
+import logging
+import re
+
+import monasca_agent.collector.checks as checks
+from monasca_agent.common.util import timeout_command
+
+
+log = logging.getLogger(__name__)
+
+_METRIC_NAME_PREFIX = "slurm"
+_METRIC_NAME = "job_status"
+
+_SLURM_LIST_JOBS_CMD = ['/usr/bin/scontrol', '-o', 'show', 'job']
+_SLURM_LIST_NODES_CMD = ['/usr/bin/scontrol', '-o', 'show', 'node']
+
+_SLURM_JOB_FIELD_REGEX = ('^JobId=([\d]+)\sJobName=((?i)[\w-]+\.[\w-]+)\sUserI'
+                          'd=([\w-]+\([\w-]+\)) GroupId=([\w-]+\([\w-]+\))\s.*'
+                          'JobState=([\w]+)\s.*\sNodeList=(.*?)\s.*$')
+_SLURM_NODE_FIELD_REGEX = '^NodeName=(.*?)\s.*State=(.*?)\s.*$'
+_SLURM_NODE_SEQUENCE_REGEX = '^(.*)-\[(\d+-\d+)\]$'
```

Comment:
Is there always a hyphen between the prefix and the numeric suffix?

Reply:
What if I have node names that are not sequential? `n1,n3` or `a,b`? Or even multiple sequences? `x[1-3],y[2-3]`

Reply:
The only examples I've seen so far have been either a single node, or a sequence of nodes. They've always had hyphens.

You're correct that it won't work if it's not sequential, but I need example output first. I'll see if I can find the slurm source code.

Reply:
The show section of the scontrol man page has some examples without hyphens.

Reply:
Good spot - looking at how the nodes are named in ```slurm.conf.j2``` the hyphen does appear to be part of the node name:

NodeName={{cluster_name}}-{{group.name}}-[0-{{group.num_nodes-1}}]